### ISSUE TITLE
Cannot install some packages in lkp-tests/distro/depends/lkp

### ISSUE THREAD

**Issue by Ankita-020696** [intel/lkp-tests#82]
Hi,
I am trying to port lkp-tests for customized distro & and I need that updated sources should work on both x86 & customized distro.
The updated sources are working fine  with customized distro but after running lkp install on ubuntu(X86) get following error,
**$ sudo lkp install
Cannot install some packages in lkp-tests/distro/depends/lkp**

How do I fix this?

Thanks

**Comment by xilabao** [comment]
@Ankita-020696 the program was stopped at below position, you can comment out the "exit 1" and install the "generic_packages" manually.
```
# grep -C1 "Cannot install some packages" lkp-exec/install
        $LKP_SRC/distro/installer/$distro $generic_packages || {
                echo "Cannot install some packages in $LKP_SRC/distro/depends/${script}"
                exit 1
```

**Comment by Ankita-020696** [comment]
Is there any generic way so that without commenting, I can use same source for both distributions?

Thanks

**Comment by Ankita-020696** [comment]
> @Ankita-020696 the program was stopped at below position, you can comment out the "exit 1" and install the "generic_packages" manually.
> 
> ```
> # grep -C1 "Cannot install some packages" lkp-exec/install
>         $LKP_SRC/distro/installer/$distro $generic_packages || {
>                 echo "Cannot install some packages in $LKP_SRC/distro/depends/${script}"
>                 exit 1
> ```

Is there any generic way so that without commenting, I can use same source for both distributions?

Thanks

**Comment by xilabao** [comment]
If you met the problem like the below one, you can add the mapping in distro/adaptation/$SYSTEM

```
commit 699cd9ff57e1935abe3dee1471df5a1dc668cb2b
Author: Tyler Hicks <tyhicks@canonical.com>
Date:   Mon Jan 6 17:31:09 2020 +0000

    distro/adaptation: which is provided by debianutils in Debian/Ubuntu
    
    Fix the following lkp install error:
    
    $ lkp install
    Use: /home/ubuntu/lkp-tests/distro/installer/ubuntu install bc gawk gzip kmod time which automake bison bsdtar|libarchive-tools build-essential bzip2 ca-certificates cpio fakeroot flex gcc git libc6-dev libc6-dev:i386 libipc-run-perl libklibc-dev libssl-dev libtool linux-libc-dev linux-libc-dev:i386 linux-tools-generic make openssl patch rsync ruby ruby-dev wget
    ...
    E: Unable to locate package which
    Cannot install some packages in /home/ubuntu/lkp-tests/distro/depends/lkp
    
    Fixes: 431fc13342bd ("distro/depends/lkp: fix "tests/wrapper: line 23: which: command not found"")
    Signed-off-by: Tyler Hicks <tyhicks@canonical.com>

diff --git a/distro/adaptation/ubuntu b/distro/adaptation/ubuntu
index 76d1ab53..c28eccd3 100644
--- a/distro/adaptation/ubuntu
+++ b/distro/adaptation/ubuntu
@@ -20,3 +20,4 @@ libgflags2v5:libgflags-dev
 perl-modules-5.24:perl-modules-5.26
 libperl5.24:libperl5.26
 pmdk:pmdk-tools
+which: debianutils
```
